clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeliverWgsCramExecutor.scala

```diff
@@ -27,16 +27,24 @@ import scala.concurrent.{ExecutionContext, Future}
 class DeliverWgsCramExecutor(deliverCommand: DeliverWgsCram)
     extends Executor[UpsertId] {
 
-  val moveCommand =
-    MoveWgsCram(deliverCommand.key, deliverCommand.workspacePath)
-  import moveCommand.index.implicits._
+  import WgsCramIndex.implicits._
 
   override def execute(webClient: ClioWebClient, ioUtil: IoUtil)(
     implicit ec: ExecutionContext
   ): Future[UpsertId] = {
 
-    val moveExecutor =
-      new MoveExecutor(moveCommand, deliverCommand.samplePrefix)
+    val deliveredFileBasename: String =
+      deliverCommand.samplePrefix.fold(deliverCommand.key.sampleAlias) {
```

Comment:
Did you consider accepting a function that could transform the destination filename, instead of the new basename? I think it would make the code here slightly simpler.

Reply:
Not sure I understand... where would the function live? Right now `deliverCommand.samplePrefix` is coming from the command line, so I think it's stuck as an `Option[String]`. We need to perform this computation somewhere, so I'm not sure wrapping it in a function to be called by the methods in the `TransferMetadata` would be any simpler than calling it once here and passing in the result.

Reply:
The function is a lambda created at the call site. Does the caller have enough information to replace the basename without needing the extension? If that could be avoided it would be much better. I don't know if we can be sure extensions of these files will never change in the future, and hardcoding them here doesn't seem great to me.

Reply:
Ah! I think I get what you're suggesting now, but I still don't know how it'd work.

In the special case of cram delivery, the caller knows that it's manipulating files that should end in ".cram", ".crai", etc, but in general that's not the case. The move-with-new-basename functionality is built into the move executor, which knows very little about the files it's operating over, and the new basename is getting parsed straight from the CLI. There's no special-cased intermediate between the parser and the executor that could transform the new basename into a lambda, and no good way to specify the lambda directly on the command line.

I managed to convince myself yesterday that the least-bad way to implement this feature is to hardcode the expected extensions. The concrete situation Sam was trying to recover from was a bug in the delivery WDL that ended up naming things like `..NA12878.cram`. I couldn't think of a sane splitting algorithm that would handle cases like this along with:

- `NA12878.cram`
- `SomePatientId.NA12878.cram`
- `NA12878.cram.crai`
- `HK3T5CCXX.6.Pond-492100.unmapped.bam`
- `NA12878.v10.bb684530-32a6-4461-a9f5-84778f61d935.g.vcf.gz`

Since I had convinced myself that the only way to tell basename apart from extension would be to hardcode the extensions somewhere, I stuck them in the transfer-metadata classes, which is where we've put all the other index-specific logic.

Reply:
But I agree there are definitely drawbacks to having things hard-coded. I built a jar from my branch and gave it to Sam to unblock him, so we can leave this open over the break and talk about it with the whole group to see if this is something we want to maintain long-term.

Reply:
Type these members not as URIs but some ExendedUri type with an extension?

Reply:
Made [DSDEGP-1827](https://broadinstitute.atlassian.net/browse/DSDEGP-1827) to track it.